v2/tools/generator/internal/astmodel/types.go

```diff
@@ -457,3 +457,42 @@ func FindStatusTypes(types Types) Types {
 
 	return result
 }
+
+// FindConnectedTypes finds all types reachable from the provided types
+// TODO: This is very similar to ReferenceGraph.Connected.
+func FindConnectedTypes(allTypes Types, roots Types) (Types, error) {
+	walker := NewTypeWalker(
+		allTypes,
+		TypeVisitorBuilder{}.Build())
+
+	result := make(Types)
+	for _, def := range roots {
+		types, err := walker.Walk(def)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed walking types")
+		}
+
+		err = result.AddTypesAllowDuplicates(types)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
+
+// FindAllSpecTypes finds all spec types and all types referenced by those spec types.
+// This differs from FindSpecTypes in that it finds not only the top level spec types but
+// also the types which the top level types are built out of.
+func FindAllSpecTypes(types Types) (Types, error) {
```

Comment:
The naming here feels misleading - it doesn't "Find all spec types" it finds "All types that are specs and all the types they reference too".

Reply:
Hmm yep - that's a good point. 

Some ideas...:
1. `FindSpecDependentTypes`
2. `FindReachableSpecTypes`
3. `FindConnectedSpecTypes` (or `FindSpecConnectedTypes`)
4. `FindSpecTypeGraph`

I think I like `FindSpecConnectedTypes`.

Reply:
`FindSpecConnectedTypes()` is pretty good.

Reply:
Fixed this